tools/not.cpp

```diff
@@ -0,0 +1,86 @@
+//===- not.cpp - The 'not' testing tool -----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Usage:
+//   not cmd
+//     Will return true if cmd doesn't crash and returns false.
+//   not --crash cmd
+//     Will return true if cmd crashes (e.g. for testing crash reporting).
+
+// This file is a stripped down version of not.cpp from llvm/utils. This does
+// not depend on any LLVM library.
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
+int main(int argc, const char **argv) {
+  bool expectCrash = false;
+
+  ++argv;
+  --argc;
+
+  if (argc > 0 && std::string(argv[0]) == "--crash") {
+    ++argv;
+    --argc;
+    expectCrash = true;
+
+    // Crash is expected, so disable crash report and symbolization to reduce
+    // output and avoid potentially slow symbolization.
+#ifdef _WIN32
+    SetEnvironmentVariableA("LLVM_DISABLE_CRASH_REPORT", "1");
+    SetEnvironmentVariableA("LLVM_DISABLE_SYMBOLIZATION", "1");
+#else
+    setenv("LLVM_DISABLE_CRASH_REPORT", "1", 0);
+    setenv("LLVM_DISABLE_SYMBOLIZATION", "1", 0);
+#endif
+  }
+
+  if (argc == 0)
+    return 1;
+
+  std::stringstream ss;
+  ss << argv[0];
+  for (int i = 1; i < argc; ++i)
+    ss << " " << argv[i];
+  std::string cmd = ss.str();
+
+  int result = std::system(cmd.c_str());
```

Comment:
I'm not sure if I'm reading this correctly, but if you include `argv[0]` -- which should be your `not` utility in this case -- in the `ss` variable, wouldn't the `std::system(cmd.c_str())` just recursively call
```bash
./not [--crash] <cmd>
```
recursively? Which would just run `not` in an infinite loop?
Again, not sure if I'm reading this correctly.